sources/osgShader/Compiler.js

```diff
@@ -1012,22 +1000,9 @@ define( [
                     continue;
                 if ( !hasShadows ) {
                     hasShadows = true;
-                    this._vertexShader.push( 'vec4 worldPosition = ModelWorldMatrix * vec4(Vertex,1.0);' );
+                    this._vertexShader.push( 'WorldPosition = ModelWorldMatrix * vec4(Vertex,1.0);' );
```

Comment:
seems risky business to have a global you *MUST* remember to reset when reusing the compiler ?
Some loops are needed anyway (gathering light/shadowtexture/shadowattribute soft linkages)